ledger-service/http-json/src/main/scala/com/digitalasset/http/json/JsonProtocol.scala

```diff
@@ -357,7 +358,43 @@ object JsonProtocol extends JsonProtocolLow {
       domain.SearchForeverRequest(NonEmptyList((single.convertTo[domain.SearchForeverQuery], 0)))
   }
 
-  implicit val CommandMetaFormat: RootJsonFormat[domain.CommandMeta] = jsonFormat3(
+  implicit val hexStringFormat: JsonFormat[HexString] =
+    jsonFormat[HexString](
+      JsonReader.func2Reader((StringJsonFormat.read _).andThen(HexString.assertFromString)),
+      JsonWriter.func2Writer[HexString](StringJsonFormat.write(_)),
+    )
+
+  implicit val deduplicationDurationTypFormat: JsonFormat[domain.DeduplicationDurationTyp] =
+    jsonFormat[domain.DeduplicationDurationTyp](
+      JsonReader.func2Reader(
+        (LongJsonFormat.read _)
+          .andThen(java.time.Duration.ofMillis)
+          .andThen(domain.DeduplicationDurationTyp(_))
+      ),
+      JsonWriter.func2Writer[domain.DeduplicationDurationTyp](duration =>
+        LongJsonFormat.write(duration.unwrap.toMillis)
+      ),
+    )
```

Comment:
There is no reason for the aforementioned `stringJsonFormat` not to work on any "underlying" `JsonFormat`, either.